packages/react-paypal-js/src/v6/hooks/usePayPalGuestPaymentSession.ts

```diff
@@ -0,0 +1,199 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { usePayPal } from "./usePayPal";
+import { useIsMountedRef } from "./useIsMounted";
+import { useError } from "./useError";
+import { useProxyProps } from "../utils";
+
+import type {
+    PayPalGuestOneTimePaymentSession,
+    PayPalGuestOneTimePaymentSessionOptions,
+    PayPalGuestOneTimePaymentSessionPromise,
+    PayPalGuestPresentationModeOptions,
+} from "@paypal/paypal-js/sdk-v6";
+import type { BasePaymentSessionReturn } from "../types";
+
+export type UsePayPalGuestPaymentSessionProps = (
+    | (Omit<PayPalGuestOneTimePaymentSessionOptions, "orderId"> & {
+          createOrder: () => PayPalGuestOneTimePaymentSessionPromise;
+          orderId?: never;
+      })
+    | (PayPalGuestOneTimePaymentSessionOptions & {
+          createOrder?: never;
+          orderId: string;
+      })
+) & {
+    buyerCountry?: string;
+    targetElement?: string | HTMLElement;
+    presentationMode?: "auto";
+    fullPageOverlay?: boolean;
+    autoRedirect?: boolean;
```

Comment:
BCDC has `inline`, `popup`, and `modal` presentation modes, so any options those can use need to be available. I think `fullPageOverlay` is valid because it's used by `popup`. I'm not sure about `autoRedirect` though, I'm only seeing that in relation to redirect presentation modes. Could you verify if this is an option BCDC needs?